### PR TITLE
chore: Create profiler catalog table and insert dummy metrics into it

### PR DESCRIPTION
**Pull request by aravindhank11**

### CODE REVIEW

File: eva/utils/profiler.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        * Create an object of classname
+        """
+        abs_path = Path(filepath).resolve()
+        spec = importlib.util.spec_from_file_location(abs_path.stem, abs_path)
+        module = importlib.util.module_from_spec(spec)
+        spec.loader.exec_module(module)
+
+        self._classobj = getattr(module, classname)
+
+    def run(self):
```
Comment (aravindhank11) [inline eva/utils/profiler.py:34]: @namandangi / @sanjeevumeshrao: This run() function needs to be modified to get the profiled metrics inserted

File: eva/catalog/catalog_manager.py [hunk 2]
```diff
@@ -284,7 +307,9 @@ def drop_udf(self, udf_name: str) -> bool:
         Returns:
            True if successfully deleted else False
         """
-        return self._udf_service.drop_udf_by_name(udf_name)
+        udf_metadata = self._udf_service.udf_by_name(udf_name)
+        profile_drop_status = self._udf_profile_service.drop_udf_profile(udf_metadata.id)
+        return profile_drop_status and self._udf_service.drop_udf_by_name(udf_name)
```
Comment (aravindhank11) [inline eva/catalog/catalog_manager.py:312]: When we drop the UDF, drop the profiled metrics as well!

File: client.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+# Drop a UDF
+cursor.execute("DROP UDF IF EXISTS FastRCNNObjectDetector")
+response = cursor.fetch_all()
+print(response)
+
+# Re-Create the UDF
+cmd = "CREATE UDF IF NOT EXISTS FastRCNNObjectDetector INPUT  (frame NDARRAY UINT8(3, ANYDIM, ANYDIM)) OUTPUT (labels NDARRAY STR(ANYDIM), bboxes NDARRAY FLOAT32(ANYDIM, 4), scores NDARRAY FLOAT32(ANYDIM)) TYPE Classification IMPL 'eva/udfs/fastrcnn_object_detector.py';"
+cursor.execute(cmd)
+response = cursor.fetch_all()
+print(response)
```
Comment (aravindhank11) [inline client.py:18]: Basic test driver. Of course, no where near comprehensive.

File: eva/catalog/services/udf_profile_service.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+        Returns:
+            UdfMetadata: Returns the new entry created
+        """
+        metadata = self.model(udf_id,
+                              metrics.batch_size,
+                              metrics.time_taken,
+                              metrics.accuracy)
+        metadata = metadata.save()
+        self.print_all_profile("Post Inserting {}".format(udf_id))
```
Comment (aravindhank11) [inline eva/catalog/services/udf_profile_service.py:42]: Unsure how to print catalog table as of now. Using print statements to ensure correctness.

File: eva/executor/create_udf_executor.py [hunk 5]
```diff
@@ -60,9 +63,19 @@ def exec(self):
...
+        udf_metadata = catalog_manager.create_udf(
             self.node.name, impl_path, self.node.udf_type, io_list
         )
+
+        # Profile the UDF
+        profiler = Profiler(impl_path, self.node.name)
+        metrics = profiler.run()
+
+        # Insert the profiled UDF to catalog
+        catalog_manager.create_udf_profile(udf_metadata.id, metrics)
```
Comment (aravindhank11) [inline eva/executor/create_udf_executor.py:77]: Profiler is blocking. TODO: Making it non-blocking

This is non-trivial: If non-blocking, potential resources are competed causing incorrect measurement of time.

Having said that, during actual execution of query as well, resources could be competed. Adaptive optimizer is the solution for this. 

NOTE: We are not intending to make it non-blocking in the scope of the project.

**Review by NetraGhaisas — APPROVED** [review]
Looks good to me

**Pull request merged**